playground/index.html

```diff
@@ -0,0 +1,110 @@
+  </div>
+  <div id="modal"
+    style="cursor: pointer; position: absolute; top: 0; left: 0; display: none; width: 100%; height: 100%; background: rgba(0,0,0,0.7);">
+    <div id="modal-inner-container"
+      style="border-radius: 12px; cursor: default; width: 90%; height: 90%; background: white; margin: 5%;">
+      <div id="modal-inner-content" style="padding: 20px;"></div>
+    </div>
+  </div>
+
+  <script>
+    function initApp() {
+      negotiatorAuthenticator.init().then(function (tokens) {
+        let isVIP = false;
+        tokens.map(ticket => { if (ticket.token.ticketClass == "VIP") isVIP = true });
+        if (isVIP) document.getElementById("vip-only-section").style.display = "block";
+        tokens.map(
+          ticketToken => {
+            const container = document.createElement("li");
+            const tokenId = document.createTextNode(ticketToken.token.ticketId);
+            container.appendChild(tokenId);
+            document.getElementById("tokenCards").appendChild(container);
+          }
+        );
+      }, function (error) {
+        console.log("error", error);
+      });
+    }
+    // to handle the vip click event
+    function vipClickEvent() {
+      // 1. Get most up to date Tokens (Confirm if this should be done)
+      negotiatorAuthenticator.init().then(function (tokens) {
+        // 2. Gather all VIP tickets
+        const vipTickets = tokens.filter(ticket => (ticket.token.ticketClass == "VIP"));
+        // 3. Build a Html template of vip tickets
+        let ticketsToHtml = "";
+        vipTickets.map(function (ticket) {
+          ticketsToHtml += `
+          <li>
+            <button style="margin: 20px" data-id="${ticket.token.ticketId}" onClick="selectVipTicket(${ticket.token.ticketId})">
+              ${ticket.token.ticketId}
+            </button>
+          </li>
+        `;
+        });
+        // 4. Render the tickets inside the vip-tickets-list
+        document.getElementById("vip-tickets-list").innerHTML = ticketsToHtml;
+        // 5. Show Modal
+        document.getElementById("vip-container").style.display = 'block';
+      }, function (error) {
+        console.log("error", error);
+      });
+    }
+    // to handle the vip selection click event from the end user
+    function selectVipTicket(ticketId) {
+      event.stopPropagation();
+      // 1. Get most up to date Tokens (Confirm if this should be done each time)
+      negotiatorAuthenticator.init().then(function (tokens) {
+        // 2. Get the ticket using ID
+        const chosenTicket = tokens.filter(ticket => (ticket.token.ticketId == ticketId))[0];
+        // 3. Authenticator (non-Disney mode) - no owner - open modal to retrieve owner
+        if (chosenTicket.ownerAddress == null) {
+          // lead to email code modal process, created by Authenticator.
+          negotiatorAuthenticator.findOwner(chosenTicket);
```

Comment:
87: findOwner should be assertOwnerAddress() - is this correct? Thanks.

Reply:
Yes that's right ('asset' is just from top of my head, anything stronger than 'find' will do)